draft-ietf-quic-qpack.md

```diff
@@ -1551,6 +1551,169 @@ the smallest number of bytes.
 | 97    | x-frame-options                  | deny                                                        |
 | 98    | x-frame-options                  | sameorigin                                                  |
 
+
+# Encoding and Decoding Examples
+
+The following examples represent a series of exchanges between an encoder and a
+decoder.  The exchanges are designed to exercise most QPACK instructions, and
+highlight potentially common patterns and their impact on dynamic table state.
+The encoder sends three encoded field sections containing one field line each,
+as wells as two speculative inserts that are not referenced.
+
+The state of the encoder's dynamic table is shown, along with its
+current size.  Each entry is shown with the Absolute Index of the entry (Abs),
+the current number of outstanding encoded field sections with references to that
+entry (Ref), along with the name and value.  Entries above the 'acknowledged'
+line have been acknowledged by the decoder.
+
+## Literal Field Line With Name Reference
+
+The encoder sends an encoded field section containing a literal representation
+of a field with a static name reference.
+
+~~~
+Data                | Interpretation
+                              | Encoder's Dynamic Table
+
+Stream: 0
+0000                | Required Insert Count = 0, Base = 0
+510b 2f69 6e64 6578 | Literal Field Line with Name Reference
+2e68 746d 6c        |  (:path=/index.html)
+
+                                Abs Ref Name        Value
+                                ^-- acknowledged --^
+                                Size=0
+~~~
+
+## Dynamic Table
+
+The encoder sets the dynamic table capacity, inserts a header with a dynamic
+name reference, then sends a potentially blocking, encoded field section
+referencing this new entry.  The decoder acknowledges processing the encoded
+field section.
+
+~~~
+Stream: Encoder
+3f8b                | Set Dynamic Table Capacity=170
+01c0 0c77 7777 2e69 | Insert With Name Reference (static table)
+6574 662e 6f72 67   |  (:authortiy=www.ietf.org)
+
+                                Abs Ref Name        Value
+                                ^-- acknowledged --^
+                                 1   0  :authority  www.ietf.org
+                                Size=54
+
+Stream: 4
+0280                | Required Insert Count = 1, Base = 0
+10                  | Indexed Field Line With Post-Base Index
+                    |  Absolute Index=1
```

Comment:
This one is useful to break down (RIC + delta_base + 0 = 1).  Shame that we have such small numbers involved here.